Jetpack performance settings

planarvoid opened the pull request:
This PR adds Jetpack performance settings based on the desings. There is now a `Site accelerator` preference which opens a screen with `Faster images` and `Faster static files` and a combined `Site accelerator` switch which turns both of them on/off. 

On the main screen you see now `Site accelerator`, `Lazy load images`, `Ad-free Video Hosting` and `More`. Clicking on `More` opens a screen where there are all the mentioned options + `Jetpack search`. 

`Ad-free video hosting` is only available for Jetpack sites with a Premium or Business plan.

`Jetpack search` is available to Jetpack sites with a business plan or WP.com sites that have the flag `isJetpackSearchSupported` set to `true`. 

The easiest way to test this is comparing the options in Calypso with the same site to the options in the WPAndroid app. It happens that the Jetpack refuses to set the settings but it happens on both platforms.

To test - with a Jetpack site with a free plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button

To test - with a Jetpack site with a premium plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above
* Try setting the new options

To test - with a Jetpack site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options

To test - with a WPCom site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options


Desings:
![jetpack_peformance_settings_android-1](https://user-images.githubusercontent.com/1079756/64017203-57e16e00-cb29-11e9-99af-00730fee907e.png)

Implementation:
![jetpack_performance_settings](https://user-images.githubusercontent.com/1079756/64017221-5f087c00-cb29-11e9-8f7a-cc2e2fd855f4.gif)

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java:
```diff
@@ -241,9 +243,26 @@
+    private WPSwitchPreference mLazyLoadImagesNested;
+
+    // Jetpack media settings
+    private WPSwitchPreference mAdFreeHosting;
```
Should we rename this one to mAdFreeVideoHosting to be consistent with naming in JetpackSettingsModel?

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/WPSwitchPreference.java:
```diff
@@ -37,6 +44,16 @@ public WPSwitchPreference(Context context, AttributeSet attrs) {
+            } else if (index == R.styleable.SummaryEditTextPreference_switchTrackTint) {
+                mTrackTint = array.getColorStateList(index);
+            } else if (index == R.styleable.SummaryEditTextPreference_preferenceTextColor) {
+                mTextColor = array.getResourceId(index, R.color.white);
```
We actually pulling this color from one of the libraries, so if it'll get removed at some point we might be in trouble. I suggest using `android.R.color.white` instead (in other places too).

planarvoid replied:
ah, I didn't know that, good point 👍

khaykov commented on WordPress/src/main/res/xml/site_settings.xml:
```diff
@@ -211,6 +190,137 @@
+                android:title="@string/site_settings_site_accelerator"
+                app:backgroundColorChecked="@color/primary_40"
+                app:backgroundColorUnchecked="@color/neutral_40"
+                app:preferenceTextColor="@color/white"
```
As I mentioned, let's switch this to `@android:color/white`

planarvoid commented:
Thanks for the review, @khaykov  !

> I see an option for "Add-free video hosting" on Personal plan, where it only starts from Premium.

I've fixed this. I thought that there are only three plans (free, premium and business) and I've checked that it's removed on the free plan. Now I'm hiding it for all the plans except premium and business. 

> I don't see the Site accelerator option for wpcom site on a business plan.

I'm checking Calypso and it doesn't make sense to me. We're setting Jetpack modules for a non-Jetpack site. Anyway, I'll do the same thing within the app. 

> Behavior of the "global" Site Accelerator switch is not consistent with Calypso - over there it stays enabled if at least one child option is selected, on Android it gets disabled unless both options are selected.

Good catch, I've fixed that.

> I know you followed a design, but the behavior of More button is a bit weird - sometimes there is nothing extra inside it, it just repeats options from previous screen. Wonder, maybe we should hide it where there is nothing extra to show?

It is weird, you're right. I'm now hiding the "More" menu when the "Improved search" is not showing (because it's the only extra item in the preferences. 

> Missing release notes :D

👍 😄
